lib/doh.c

```diff
@@ -423,7 +423,7 @@ struct Curl_addrinfo *Curl_doh(struct Curl_easy *data,
     dohp->pending++;
   }
 
-  if(conn->ip_version != CURL_IPRESOLVE_V4) {
+  if(data->set.ipver != CURL_IPRESOLVE_V4) {
```

Comment:
Isn't this commit rather a bug-fix that is independent of your "strict IP version" work? If so, I would prefer to handle it in a separate PR.

Reply:
Oh, I forgot this change when I refactored to from the first version with the new setting, to this new version using the `_STRICT` options. They are only relevant to the first version.

Please let me know you like this change by itself, so I can submit the commit in a different PR. Otherwise I'll just remove it.